files/bin/deploy-iqe-cji.py

```diff
@@ -53,23 +54,23 @@ def job_name(self) -> str:
         return self.pipeline_run_name.rsplit("-", 1)[0]
 
     @cached_property
-    def build_number(self) -> str:
-        """Use the first five digits of CHECK_RUN_ID.
+    def get_run_identifier(self) -> str:
+        """Return the CHECK_RUN_ID used to identify this run.
 
-        Default to 1 if CHECK_RUN_ID is unset or falsy value.
+        If CHECK_RUN_ID is unset or falsy, return a short, base64-encoded random string.
 
         Example:
-            31510716818 --> 31510
+            CHECK_RUN_ID=31510716818 --> "31510716818"
+            CHECK_RUN_ID not set     --> "c91a0f3d-4dbe-4b3b-9e5d-92b542f9f9f7"
         """
+        check_run_id = os.environ.get("CHECK_RUN_ID")
+        return str(check_run_id) if check_run_id else uuid.uuid4().hex[:8]
```

Comment:
Now that I understand how this is used, it no longer makes sense to have a backup value here or in the other file, unless you can guarantee they are the same. Whatever value we use for the deploy needs to match what we give to IQE. using uuid4 will ensure the values will be different.

Reply:
I updated this to parse the hash from the pipeline run id instead.